Voice tutorial:  Add procedure on setting up esp32 for voice control

c0ffeeca7 opened the pull request:
## Proposed change
<!-- 
    Describe the big picture of your changes here to communicate to the
    maintainers why we should accept this pull request. If it fixes a bug
    or resolves a feature request, be sure to link to that issue in the 
    additional information section.
-->
Assist tutorials:
- Add procedure on sestting up esp32 for voice control


## Type of change
<!--
    What types of changes does your PR introduce to our documentation/website?
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR.
-->

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [x] Added documentation for a new integration others are adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information
<!--
    Details are important, and help maintainers processing your PR.
    Please be sure to fill out additional details, if applicable.
-->

- Link to parent pull request in the codebase: 
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist
<!--
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR. If you're unsure about any of them, don't hesitate to ask.
    We're here to help! This is simply a reminder of what we are going to look
    for before merging your code.
-->

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

c0ffeeca7 committed:
Assist: add procedure on setting up ESP32 for voice control

- Draft

c0ffeeca7 committed:
Assist ESP32: Add ref to troubleshooting section

balloob commented on source/projects/private-voice-assistant/voice_remote_esp32.markdown:
```diff
@@ -0,0 +1,58 @@
+---
+title: "Setting up an ESP32 to control Home Assistant"
```
Should we make it more clickbaity?

```suggestion
title: "$13 voice remote that controls Home Assistant"
```

balloob commented on source/projects/private-voice-assistant/voice_remote_esp32.markdown:
```diff
@@ -0,0 +1,58 @@
+* [Home Assistant Cloud](https://www.nabucasa.com) or a manually configured [Assist Pipeline](/integrations/assist_pipeline)
+* The password to your 2.4&nbsp;GHz Wi-Fi network
+* Chrome (or a Chromium-based browser like Edge) on desktop (not Android/iOS) 
+* ESP32 board with a built-in speaker: [M5Stack ATOM Echo Development Kit](https://shop.m5stack.com/products/atom-echo-smart-speaker-dev-kit?ref=NabuCasa)
```
Probably just mention the product right away
```suggestion
* [M5Stack ATOM Echo Development Kit](https://shop.m5stack.com/products/atom-echo-smart-speaker-dev-kit?ref=NabuCasa)
```

balloob commented on source/projects/private-voice-assistant/voice_remote_esp32.markdown:
```diff
@@ -0,0 +1,58 @@
+## Required material
+
+* Latest version of Home Assistant installed on your Home Assistant device 
+* A controllable device in your Home Assistant network
```
I would leave this one out, seems obvious? 
```suggestion
```

balloob commented on source/projects/private-voice-assistant/voice_remote_esp32.markdown:
```diff
@@ -0,0 +1,58 @@
+   <script type="module" src="https://unpkg.com/esp-web-tools@9/dist/web/install-button.js?module"></script>
+   <esp-web-install-button manifest="https://firmware.esphome.io/voice-assistant/m5stack-atom-echo/manifest.json"></esp-web-install-button>
+
+   * In the pop-up window, select the USB serial port and select **Connect**.
```
It's not clear sometimes what USB serial port belongs to the device. To make it obvious, let users plug in the device after the window is open, and they see the serial port appear. That's the one they should pick.

balloob commented on source/projects/private-voice-assistant/voice_remote_esp32.markdown:
```diff
@@ -0,0 +1,58 @@
+        * Select **Install Voice Assistant**, then **Install**.
+        * Follow the instructions provided by the installation wizard.
+        * When prompted, enter the credentials to your 2.4&nbsp;GHz Wi-Fi network.
+          * As the **Network name**, enter the SSID of your 2.4&nbsp;GHz Wi-Fi.
```
There should be a drop down to pick your network, unless you select "Other". It can sometimes be that the device doesn't have enough time to scan for networks.

balloob commented on source/projects/private-voice-assistant/voice_remote_esp32.markdown:
```diff
@@ -0,0 +1,58 @@
+1. In Home Assistant, go to {% my integrations title="**Settings** > **Devices & Services**" %}.
+   * You should now see a new **M5Stack Atom Echo** integration.
+   * Select **Configure** and then **Submit**.
+   * The ESP32 integration should now show all the entities.
```
```suggestion
   * The ESPHome config entry should now show all the entities.
```

balloob commented on source/projects/private-voice-assistant/voice_remote_esp32.markdown:
```diff
@@ -0,0 +1,58 @@
+1. Congratulations! You've setup your ATOM Echo device to voice control Home Assistant. Now give some commands.
+   * Press the button on your ATOM Echo.
+     * The LED should light up in blue.
+     * Say a [supported voice command](/docs/assist/builtin_sentences/). For example, *Turn off the light in the kitchen*.
```
Just a note, that we default to the language configured in Home Assistant, which might not be English.

balloob commented on source/projects/private-voice-assistant/voice_remote_esp32.markdown:
```diff
@@ -0,0 +1,58 @@
+     * You can also ask a question, such as
+        * *Is the front door locked?*
+        * *Which lights are on in the living room?*
+        * Make sure you’re using the area name as you defined it in Home Assistant. If you have a room called bathroom, the phrase *Turn on the lights in the bath* won’t work.
```
This seems a bit too specific to be listed here? Probably better in a generic voice command docs. Also, good to note in that case that we support area aliases to cover this.

balloob commented on source/projects/private-voice-assistant/voice_remote_esp32.markdown:
```diff
@@ -0,0 +1,58 @@
+   * Let go of the button.
+     * The LED should light up in green.
+     * Home Assistant will confirm the action.
+1. Your command was not recognized? Test if the command works:
```
Should this be under troubleshooting?

c0ffeeca7 committed:
Apply suggestions from code review

Co-authored-by: Paulus Schoutsen <balloob@gmail.com>

c0ffeeca7 committed:
ESP32 voice remote: update procedure

c0ffeeca7 committed:
Merge branch 'next' into voice-tutorial-esp32

c0ffeeca7 committed:
Fix typos

c0ffeeca7 committed:
Add link to procedure on how to create custom sentences

c0ffeeca7 committed:
Add step on installing driver

c0ffeeca7 committed:
Implement review feedback by ballob

c0ffeeca7 committed:
Fix typo

frenck reviewed (dismissed):
Looks good to me, couldn't fine anything out of the ordinary 👍 

../Frenck

c0ffeeca7 committed:
Merge branch 'next' into voice-tutorial-esp32

c0ffeeca7 committed:
Add some more procedures

- Add procedure on creating a voice remote if no Cloud subscription is available
- add procedure on creating a local assistant using Whisper and Piper
- add procedure on exposing entities to Assist

c0ffeeca7 committed:
Minor tweaks

balloob commented on source/_includes/asides/docs_navigation.html:
```diff
@@ -46,6 +46,10 @@ <h1 class="title delta">Topics</h1>
           <li>{% active_link /docs/assist/custom_sentences/ Custom sentences %}</li>
+          <li>{% active_link /projects/private-voice-assistant/voice_remote_esp32/ Setting up a $13 voice remote to control Home Assistant %}</li>
+          <li>{% active_link /projects/private-voice-assistant/voice_remote_local_device/ Setting up a voice remote that processes locally %}</li>
+          <li>{% active_link /projects/private-voice-assistant/voice_remote_local_assistant/ Configuring a local assistant %}</li>
+          <li>{% active_link /projects/private-voice-assistant/voice_remote_expose_devices/ Exposing devices to your voice assistant %}</li>
```
These 2 should be under docs. They are not projects.

balloob commented on source/_includes/asides/docs_navigation.html:
```diff
@@ -46,6 +46,10 @@ <h1 class="title delta">Topics</h1>
           <li>{% active_link /docs/assist/apple/ Assist for Apple devices %}</li>
           <li>{% active_link /docs/assist/builtin_sentences/ Built-in sentences %}</li>
           <li>{% active_link /docs/assist/custom_sentences/ Custom sentences %}</li>
+          <li>{% active_link /projects/private-voice-assistant/voice_remote_esp32/ Setting up a $13 voice remote to control Home Assistant %}</li>
+          <li>{% active_link /projects/private-voice-assistant/voice_remote_local_device/ Setting up a voice remote that processes locally %}</li>
```
These 2 are projects but are linked in the docs navigation, should they just be under docs?

Also, the analog phone is the World's Most Private Voice Assistant. Currently it has 2 folders `private-voice-assistant/voice_remote_local_device/`. Let's change this to `/projects/worlds-most-private-voice-assistant`

The ESP tutorial is the "$13 voice remote for Home Assistant". It currently also has 2 folders. Let's change this to `/projects/13-usd-voice-remote`

balloob commented on source/projects/private-voice-assistant/voice_remote_esp32.markdown:
```diff
@@ -0,0 +1,81 @@
+---
+title: "Setting up a $13 voice remote to control Home Assistant"
```
```suggestion
title: "$13 voice remote to control Home Assistant"
```

c0ffeeca7 replied:
I'm using a system here. The -ing form is a visual cue telling readers that this is a task topic. A procedure that tells them step-by-step how to do something. So if they are looking for a procedure they can ignore anything that does not start with -ing. 

But I get that you are probably viewing this more from a marketing perspective.

balloob commented on source/projects/private-voice-assistant/voice_remote_esp32.markdown:
```diff
@@ -0,0 +1,81 @@
+world's most private voice assistant. Pick up the tiny device to talk to
+your smart home. Issue commands and get responses!
+
+* Don't have Home Assistant Cloud? You just want fully local processing? Follow this procedure: [Setting up a fully local voice remote device](/projects/private-voice-assistant/voice_remote_local_device/)
```
We should link this video instead:

```suggestion
<lite-youtube videoid="w6QxGdxVMJs" videotitle="$13 voice remote for Home Assistant
"></lite-youtube>
```

The local pipeline suggestion should be on the `assist_pipeline` docs, which will be linked in requirements.

balloob commented on source/projects/private-voice-assistant/voice_remote_esp32.markdown:
```diff
@@ -0,0 +1,81 @@
+## Required material
+
+* Latest version of Home Assistant installed on your Home Assistant device 
+* [Home Assistant Cloud](https://www.nabucasa.com)
```
```suggestion
* [Home Assistant Cloud](https://www.nabucasa.com) or a manually configured [Assist Pipeline](/integrations/assist_pipeline)
```

balloob commented on source/projects/private-voice-assistant/voice_remote_esp32.markdown:
```diff
@@ -0,0 +1,81 @@
+
+## Required material
+
+* Latest version of Home Assistant installed on your Home Assistant device 
```
```suggestion
* Home Assistant 2023.5 or later
```

balloob commented on source/projects/private-voice-assistant/voice_remote_esp32.markdown:
```diff
@@ -0,0 +1,81 @@
+       * The ATOM Echo now joined your network. Select **Add to Home Assistant**.
+1. This opens the **My** link to Home Assistant. 
+   * If you haven't used My links before and previously changed the hostname, in `http://homeassistant.local:8123` replace `homeassistant` with the hostname or IP address of your local Home Assistant instance.
+   * Select **Save**, then **Open link**.
```
"Select Save" is only applicable if people haven't used My links before

balloob commented on source/projects/private-voice-assistant/voice_remote_expose_devices.markdown:
```diff
@@ -0,0 +1,15 @@
+---
+title: "Expose devices"
```
This should not be under projects. This is generic documentation that applies to Assist, Google and Amazon. I suggest this is moved to it's own PR.

balloob commented on source/projects/private-voice-assistant/voice_remote_local_assistant.markdown:
```diff
@@ -0,0 +1,35 @@
+## Setting up a local voice assistant
+
+1. Install the add-ons to convert text into speech and vice versa.
+   * Under **Settings** > **Add-ons** > **Z-Wave JS**, install the {% my supervisor_addon addon="whisper" title="**Whisper**" %} and the {% my supervisor_addon addon="piper" title="**Piper**" %} 
```
Z-Wave JS?

balloob replied:
Also, you don't need to go to settings/add-ons if you click a My link, as that deeplinks to the right location.

c0ffeeca7 replied:
I know. I'm always mentioning how to get there because I used My links a lot. The problem was then (and still is) for me that this way I don't learn where things are located within HA. And if I then look for something while I'm not coming from the documentation I struggle to find things. 
This way I can click on the My link for convenience but I still see the path where this is taking me.

balloob commented on source/projects/private-voice-assistant/voice_remote_local_assistant.markdown:
```diff
@@ -0,0 +1,35 @@
+---
+title: "Configuring a local voice assistant"
```
These instructions should be moved into it's own PR. They are also not related to the `projects` but should be generic Assist documentation page (or maybe live on the assist pipeline docs?)

balloob replied:
```suggestion
title: "Configuring a local assist pipeline"
```

balloob commented on source/projects/private-voice-assistant/voice_remote_local_assistant.markdown:
```diff
@@ -0,0 +1,35 @@
+      ![Install the Whisper and Piper add-ons](/images/assist/piper-whisper-install-01.png)
+   * Start both add-ons. This may take a while.
+   * Once the add-ons are started, head over to the integrations under {% my integrations title="**Settings** > **Devices & Services**" %}.
+      * You should now see both integrations.
```
```suggestion
      * You should now see both being discovered by the [Wyoming integration](/integations/wyoming).
```

balloob commented on source/projects/private-voice-assistant/voice_remote_local_device.markdown:
```diff
@@ -0,0 +1,38 @@
+---
+title: "Setting up a fully local voice remote device"
```
Let's remove this file. We have instructions on how to create a local pipeline and instructions on how to setup a device to talk to pipelines.

balloob commented on source/projects/private-voice-assistant/voice_remote_local_assistant.markdown:
```diff
@@ -0,0 +1,35 @@
+title: "Configuring a local voice assistant"
+---
+
+If you have Home Assistant Cloud, the voice assistant uses Home Assistant Cloud by default.
+This is because the Cloud version currently still performs better than the local version.
+The performance depends a bit on the hardware you run it on. To set up a fully local, private voice assistant, follow these steps.
```
```suggestion
In Home Assistant, the Assist pipelines are made up of various components that together form a voice assistant.

For each component you can choose from different options. We have prepared a speech-to-text and text-to-speech option that runs fully local.

The speech-to-text option is [Whisper](https://github.com/openai/whisper). It's an open source AI model that supports [various languages](https://github.com/openai/whisper#available-models-and-languages). We use a forked version called [faster-whisper](https://github.com/guillaumekln/faster-whisper). On a Raspberry Pi 4, it takes around 8 seconds to process incoming voice commands. On an Intel NUC it is done in under a second.

For text-to-speech we have developed [Piper](https://github.com/rhasspy/piper). Piper is a fast, local neural text to speech system that sounds great and is optimized for the Raspberry Pi 4. It supports [many languages](https://rhasspy.github.io/piper-samples/). On a Raspberry Pi, using medium quality models, it can generate 1.6s of voice in a second.
```

c0ffeeca7 committed:
Merge branch 'next' into voice-tutorial-esp32

c0ffeeca7 committed:
Apply suggestions from code review

Co-authored-by: Paulus Schoutsen <balloob@gmail.com>

c0ffeeca7 committed:
Implement review feedback

c0ffeeca7 committed:
Implement review feedback

c0ffeeca7 committed:
Fix typo

c0ffeeca7 committed:
Remove instructions on creating fully local esp32 setup

c0ffeeca7 committed:
Delete step to check server language

c0ffeeca7 committed:
Remove dead link in aside

c0ffeeca7 committed:
Fix link

c0ffeeca7 committed:
Improve wording

c0ffeeca7 committed:
Improve wording

c0ffeeca7 committed:
Fix broken link

c0ffeeca7 committed:
Implement review feedback

c0ffeeca7 committed:
Fix navigation text
